### PR TITLE
Scaling XGBoost example

### PR DESCRIPTION
**Pull request by jrbourbeau**
This PR adds an example notebook to demonstrate using Dask + Coiled to scale XGBoost. 

@mrocklin some notes on the dataset currently being used:

- I've initially gone with a generated dataset from `sklearn.datasets.make_classification` which is less exciting than a real dataset but enables us to nicely motivate scaling out by generating a 32GB dataset

- Initially I had wanted to use https://archive.ics.uci.edu/ml/datasets/HIGGS, but it's only 7.5 GB and not available on S3, so I would have to ship the dataset to the same filesystem that the workers are on

- What do you think about switching to the NYC taxi dataset on S3 and making this a regression problem instead of binary classification?

### CODE REVIEW

**Comment by mrocklin** [comment]
> Initially I had wanted to use https://archive.ics.uci.edu/ml/datasets/HIGGS, but it's only 7.5 GB and not available on S3, so I would have to ship the dataset to the same filesystem that the workers are on

We could host this on s3://coiled-data ?

**Comment by mrocklin** [comment]
cc also @stsievert who might have thoughts on how best to show off XGBoost

**Comment by stsievert** [comment]
I'd look at Kaggle for a good example problem. That's where gradient boosted trees have seen a lot of success: I think they win something like 50% of all the contests.

To me a 7.5GB dataset sounds large. MNIST/CIFAR10 are ~200MB/700MB *after* image transformations respectively. Typically image transforms are only done right before the prediction.

> What do you think about switching to the NYC taxi dataset on S3 and making this a regression problem instead of binary classification?

Does regression with XGBoost make sense? If I understand XGBoost correctly, each tree outputs a discrete value. It looks like the output of the forest is the sum/mean/median of all the trees: https://homes.cs.washington.edu/~tqchen/pdf/BoostedTree.pdf

**Comment by jrbourbeau** [comment]
> To me a 7.5GB dataset sounds large

That's good to know. Unfortunately that 7.5GB number I gave was for the size of the CSV file on disk. When loaded into memory the Higgs dataset is 2.5 GB. 

> We could host this on s3://coiled-data ?

We now have CSV files in `s3://coiled-data/higgs` and I've updated the notebook here to use that dataset instead of `make_classification`

I think the notebook is in a better place now than it was a couple of days ago. Feel free to take another look whenever you get a chance. Feedback is very welcome

**Comment by mrocklin** [comment]
I tried this locally and ran into a few questions:

1.  We'll have to figure out where to place a cluster configuration.  Maybe coiled/xgboost?
2.  Bokeh didn't work for me, I had to install jupyter_bokeh .  Maybe switch to matplotlib instead to avoid this for others?  https://discourse.bokeh.org/t/javascript-error-property-grid-axis-wasnt-declared/5552
3.  Is the area under the ROC curve a measure of success?  If so, training on the full dataset with these parameters doesn't seem to improve performance at all.  We might look around for a better success story.

**Comment by jrbourbeau** [comment]
> We'll have to figure out where to place a cluster configuration

I've already created a `coiled-examples` account and it has a `coiled-examples/xgboost` cluster configuration we're currently using in this notebook. I opted for `coiled-examples` instead of `coiled` as coiled's cluster configurations are listed for all users on the web UI and we'll eventually have many of them 

> Bokeh didn't work for me, I had to install jupyter_bokeh . Maybe switch to matplotlib instead to avoid this for others?

Ah yes. Sorry about that, I probably installed `jupyter_bokeh` locally. Will switch to matplotlib 👍 

> Is the area under the ROC curve a measure of success?

Yes, but I suppose the first CSV file has enough data that we don't get any gains in the AUC by training on more data. I'll think on this point to see about getting a better success story

**Comment by jrbourbeau** [comment]
I've updates the notebook to use matplotlib and have uploaded the `coiled-examples/xgboost` cluster configuration to beta.coiledhq.com. I'm going to merge this PR to get this example out, but will revisit your AUC comment in a follow-up PR. 

Thanks for reviewing @stsievert @mrocklin!
